java/src/org/openqa/selenium/devtools/idealized/Network.java

```diff
@@ -200,6 +200,12 @@ public void prepareToInterceptTraffic() {
         pausedRequest -> {
           try {
             String id = getRequestId(pausedRequest);
+
+            if (hasErrorResponse(pausedRequest)) {
+              devTools.send(continueWithoutModification(pausedRequest));
```

Comment:
Do we need to clean up the future in pendingResponses for this interception?

Reply:
Yes, you are right. I have updated the PR for the same.